### PR TITLE
Elm018

### PR DESCRIPTION
**Pull request by mradke**
Hey thanks for this helpful example repository.
I just updated all dependencies to the latest version and refactored the Main.elm file quite a bit (hoping to adjust it as far as possible to best practices).

Please let me know if you want or need any changes to be applied in addition to what I already did before the merge.

### CODE REVIEW

**Comment by mattjbray** [comment]
A couple of observations:

* `elm-export` and `servant-elm` are now both in Stackage, so @mradke you should be able to remove all references to them from `stack.yaml`.

* @soenkehahn `elm-decode-pipeline` is actually required by the decoders `elm-export` generates, so adding this dependency is directly related to this version update.

File: server/src/Api.hs [hunk 1]
```diff
@@ -23,11 +23,11 @@ api = Proxy
 
 -- types
 
-type ItemId = Integer
+type ItemId = Int
```
Comment (mradke) [inline server/src/Api.hs]: I always got an error when using Integer. I assume that I did not correctly derived the type previously. Will have a closer look at that.

**Comment by mradke** [comment]
> I've left some comments on the first commit. But most importantly: Did you run the test-suite? Does it pass for you? It doesn't for me...
Well no. Sadly not... Will try to fix that asap, but I fear I might need to reach out to you again to do so (since my experience testing Haskell is limited to say the best) 🙃 

> For example starting to use elm-decode-pipeline -- while reasonable -- is not tied to updating to a new elm version, if I understand it right.

As @mattjbray already pointed out (thanks for chipping in! 👍) this is in deed necessary.

> Also, it's nice to have separate smaller PRs for things that can be discussed separately. 
Yeah you are probably right. I'd suggest to close this request and split it to two:

* Update the server dependencies (and making the test suite pass again)
* Updating the client code seperately

If you agree, please just close the request and I'll bug you again in a few days :)

**Comment by soenkehahn** [comment]
Yeah, it'd be nice to split it up into server changes and client changes. I assumed that wasn't possible, since the code generation script probably generates code that only works with one version of `elm`. If that's not the case, all the better.

Closing this for now, as per your request.

**Pull request closed without merging**